File: tests/fixtures/kotlin/scope.kt
package fixtures.scope

import vendor.Worker as ImportedWorker

class Worker(val name: String)

val callback = { value: String -> normalize(value) }
val eager = ImportedWorker.create()

fun helper(value: String): Worker = Worker(value)

fun run(factory: () -> Worker) {
  val local = { name: String -> helper(name) }
  factory()
  callback("one")
  local("two")
  helper("three")
  ImportedWorker.create()
  missing()
}

File: tests/fixtures/lua/scope.lua
local function helper(value)
  return normalize(value)
end

local callback = function(item)
  return helper(item)
end

local first, second = loadFirst(), loadSecond()

function run(loader)
  local local_callback = function(value)
    return callback(value)
  end

  loader()
  callback(first)
  local_callback(second)
  missing()
end
